attributes/geo:proj/README.md

```diff
@@ -0,0 +1,186 @@
+1. Identifying the spatial dimensions using either `spatial_dimensions` or convention-based detection
+2. Looking up these dimension names in the Zarr array's `dimension_names`
+3. Using the corresponding sizes from the array's `shape` attribute
+
+This approach avoids redundancy and ensures consistency by using the array's own metadata rather than duplicating shape information.
+
+## Examples
+
+### Example 1: Simple EPSG Code
+
+```json
+{
+  "zarr_format": 3,
+  "shape": [2048, 2048],
+  "dimension_names": ["y", "x"],
+  "attributes": {
+    "geo:proj": {
+      "code": "EPSG:3857"
+    }
+  }
+}
+```
+
+### Example 2: With Multiple Dimensions and Transform
+
+```json
+{
+  "zarr_format": 3,
+  "shape": [365, 100, 2048, 2048, 4],
+  "dimension_names": ["time", "height", "latitude", "longitude", "band"],
+  "attributes": {
+    "geo:proj": {
+      "code": "EPSG:4326",
+      "spatial_dimensions": ["latitude", "longitude"],
+      "transform": [0.1, 0.0, -180.0, 0.0, -0.1, 90.0],
+      "bbox": [-180.0, -90.0, 180.0, 90.0]
+    }
+  }
+}
+```
+
+### Example 3: Irregular Grid
+
+In this case, the spatial coordinates are stored in separate coordinate variables (`x` and `y`) that share the same dimension as the data array:
+
+```json
+{
+  "zarr_format": 3,
+  "shape": [1000],
+  "dimension_names": ["node"],
+  "attributes": {
+    "geo:proj": {
+      "code": "EPSG:4326",
+      "spatial_dimensions": ["node"]
+    }
+  },
+  "coordinates": {
+    "x": {
+      "shape": [1000],
+      "dimension_names": ["node"]
+    },
+    "y": {
+      "shape": [1000],
+      "dimension_names": ["node"]
+    }
+  }
+}
+```
+
+Note: The coordinate variables contain the actual spatial positions in the CRS for each node.
+
+### Example 4: WKT2 Representation
+
+```json
+{
+  "zarr_format": 3,
+  "shape": [1000, 1000],
+  "dimension_names": ["northing", "easting"],
+  "attributes": {
+    "geo:proj": {
+      "wkt2": "PROJCRS[\"WGS 84 / UTM zone 33N\",BASEGEOGCRS[\"WGS 84\",DATUM[\"World Geodetic System 1984\",ELLIPSOID[\"WGS 84\",6378137,298.257223563,LENGTHUNIT[\"metre\",1]]],PRIMEM[\"Greenwich\",0,ANGLEUNIT[\"degree\",0.0174532925199433]]],CONVERSION[\"UTM zone 33N\",METHOD[\"Transverse Mercator\",ID[\"EPSG\",9807]],PARAMETER[\"Latitude of natural origin\",0,ANGLEUNIT[\"degree\",0.0174532925199433]],PARAMETER[\"Longitude of natural origin\",15,ANGLEUNIT[\"degree\",0.0174532925199433]],PARAMETER[\"Scale factor at natural origin\",0.9996,SCALEUNIT[\"unity\",1]],PARAMETER[\"False easting\",500000,LENGTHUNIT[\"metre\",1]],PARAMETER[\"False northing\",0,LENGTHUNIT[\"metre\",1]]],CS[Cartesian,2],AXIS[\"easting\",east,ORDER[1],LENGTHUNIT[\"metre\",1]],AXIS[\"northing\",north,ORDER[2],LENGTHUNIT[\"metre\",1]]]",
+      "transform": [30.0, 0.0, 500000.0, 0.0, -30.0, 5000000.0]
+    }
+  }
+}
+```
+
+## Inheritance
+
+When `geo:proj` is defined at the group level, it applies to all arrays within that group unless overridden at the array level.
```

Comment:
At which point does inheritance stops? Would this extension allow a group to override the attribute defined in its parent group or any attribute found when traversing the hierarchy in the child->parent direction? 

Is partial inheritance allowed? I.e., override only a subset of the attribute fields in arrays or subgroups. I guess better would be to disallow it! At least for mow... I'm wondering if there would be cases in the future where this would make sense, especially if this extension gather different kinds of geo-referencing metadata (CRS, transforms... GCPs?, etc.).

Reply:
I actually meant partial inheritance group->array and not general parent->child in the whole hierarchy. I don't know how to describe that preceisely. I will make a try